### DISCUSSION TITLE
How can I decode a Bucket value?

### DISCUSSION THREAD

**Discussion by aarya-brex** [vladimir-bukhtoyarov/bucket4j#490] [General]
I'm not really finding any documentation on what is actually being written to Redis, this is what I have:
```bash
redis:6379> GET REDACTED
"\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00'\x10\x00\x00\x00\x00\x00\x00'\x10\x00\x00N\x94\x91O\x00\x00\x00\x00\x00\x00\x00\x00'\x10\x01\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x01\x00\x00\x00\x03\x17\xcd\x81\xff\xe1\x81\x06\x80\x00\x00\x00\x00\x00\x00'\x0f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x04"
```

This is used to build a monitoring dashboard, so being able to decode this in Javascript is ideal

**Comment by vladimir-bukhtoyarov** [comment]
Hello @aarya-brex 

> I'm not really finding any documentation on what is actually being written to Redis

It is not a part of public API, so this will never be documented.

You can deserialize array of bytes via this code [InternalSerializationHelper#deserializeState](https://github.com/bucket4j/bucket4j/blob/0a843dce4ce94748cfe936d5903e644b80747248/bucket4j-core/src/main/java/io/github/bucket4j/distributed/serialization/InternalSerializationHelper.java#L47)

**Comment by aarya-brex** [comment]
This is a great place to start, thanks! I will see if I can figure out from here and will post my solution here

**Reply by aarya-brex** [reply]
For the sake of completion, this is what worked for me to get availableTokens:
```javascript
let textEncoder = new TextEncoder();
let longInteger = BigInt(0);
textEncoder
  .encode(encodedValue)
  .slice(-8)
  .forEach(
    (index) => (longInteger = (longInteger << BigInt(8)) | BigInt(index))
  );
return Number(longInteger);
```
